### ISSUE TITLE
How to detect when the idle_time_out has been reached?

### ISSUE THREAD

**Issue by chradek** [amqp/rhea#264]
**Describe the bug**
I'm creating a receiver and have set the `idle_time_out` as one of my connection options. When the idle timeout is reached, I can see "max idle time exceeded" in the logs (rhea.frames) but none of my event listeners on the connection or receiver get triggered. I'd expect `connection_close` to be fired.

**To Reproduce**
 - Run the below script in a linux environment.
 - Observe messages being received.
 - Use iptables to block incoming traffic on the local port communicating to the service (`iptables -A INPUT -p tcp --destination-port <PORT> -j DROP`)
 - Observe that the idle_time_out is reached via the rhea logs.

Example script:
```javascript
const {
  Connection,
  ConnectionEvents,
  ReceiverEvents,
  types
} = require('rhea-promise');

const host = process.env.AMQP_HOST;
const username = process.env.AMQP_USERNAME;
const password = process.env.AMQP_PASSWORD;
const address = process.env.RECEIVER_ADDRESS;

async function run() {
  const connectionOptions = {
    transport: "tls",
    host: host,
    hostname: host,
    username: username,
    port: 5671,
    reconnect: false,
    password: password,
    idle_time_out: 10000
  };

  const connection = new Connection(connectionOptions);

  for (const type of Object.keys(ConnectionEvents)) {
    connection.on(type, (...args) => {
      console.log(`Connection event ${type} triggered with`, ...args);
    });
  }

  const receiverName = `receiver=${Date.now()}`;
  const filterClause = `amqp.annotation.x-opt-enqueued-time > '${Date.now() - 3600 * 1000}'`;

  const receiverOptions = {
    name: receiverName,
    source: {
      address: address,
      filter: {
        "apache.org:selector-filter:string": types.wrap_described(filterClause, 0x468C00000004)
      }
    },
    onSettled: (context) => {
      console.log(`settled`);
    },
    onSessionError: (context) => {
      console.log('session error');
    },
    onClose: (context) => {
      console.log(`receiver close`);
    },
    onSessionClose: (context) => {
      console.log(`session_close`);
    }
  };

  await connection.open();

  const receiver = await connection.createReceiver(receiverOptions);

  receiver.on(ReceiverEvents.message, (context) => {
    console.log('message received');
  });
  receiver.on(ReceiverEvents.receiverError, (context) => {
    const error = context.receiver && context.receiver.error;
    console.log('receiver error', error);
  });

  setInterval(() => {
    console.log(`Is connection open?: ${connection.isOpen()}`);
    console.log(`Is remote open?: ${connection.isRemoteOpen()}`);
  }, 2000);
}

run();
```

**Expected behavior**
I'd expect an event handler on the connection to be invoked.

It looks like I can detect when the local connection is closed via `connection.isOpen()`, which will be `false`, but that requires polling the status of the connection.

**Package-version**: 
- rhea-promise: 1.0.0
- rhea: 1.0.13

**node.js version**:
12.3.1

**OS name and version**:
Debian 9 (in docker)

**Additional context**
Wasn't sure if this belonged in `rhea` or `rhea-promise`. The `Connection.idle` method is getting invoked but doesn't appear to dispatch any events, so this seems like the right repo for this issue.
https://github.com/amqp/rhea/blob/f2ef8a34e13d1933654147c5d790bf0bcb3b3240/lib/connection.js#L529-L535

/cc @ramya-rao-a

**Comment by grs** [comment]
Thanks! Fixed in 1.0.14

**Comment by chradek** [comment]
Thanks for the quick turnaround! Confirmed I now see the disconnected event when I upgrade to 1.0.14.
